060-kafka-roller-kraft.md

```diff
@@ -0,0 +1,159 @@
+# Kafka Roller KRaft Support
+
+This proposal describes the actions that the KafkaRoller should take when operating 
+against a Strimzi cluster in KRaft mode.
+The proposal describes the checks the KafkaRoller should take, how it should perform 
+those checks and in what order but does not discuss exactly how the KafkaRoller works.
+This proposal is expected to apply to both the current KafkaRoller and a future iteration of the KafkaRoller.
+
+This proposal assumes that liveness/readiness of nodes is as described in [proposal #46](https://github.com/strimzi/proposals/blob/main/046-kraft-liveness-readiness.md) 
+and [PR #8892](https://github.com/strimzi/strimzi-kafka-operator/pull/8892).
+
+
+## Current situation
+
+When operating on a ZooKeeper based cluster the KafkaRoller has the behaviour 
+described below.
+
+### Order to restart nodes
+Restart in order:
+ 1. Unready brokers
+ 2. Ready brokers
+ 3. Current controller
+
+### Triggers
+The following are some of the triggers that restart a broker:
+- Pod or its StrimziPodSet is annotated for manual rolling update
+- Pod is unready
+- Broker's configuration has changed
+
+### Restart conditions
+ 1. Does not restart a broker performing log recovery
+ 2. Attempts to connect an admin client to a broker and if it can't connect, restarts the broker
+ 3. Restarts a broker if the pod is in stuck state
+ 4. Restarts a broker if the configuration has changed and cannot be updated dynamically
+ 5. Does not restart a broker if doing so would take the in-sync replicas count below `min.insync.replicas` for any topic hosted on that broker
```

Comment:
KafkaRoller checks these conditions in order. Instead of splitting out into 2 lists, we added a sentence to clarify that the restart conditions are checked in the specified order.